source/fuzz/transformation_vector_shuffle.cpp

```diff
@@ -151,46 +155,51 @@ void TransformationVectorShuffle::Apply(
 
   // Add synonym facts relating the defined elements of the shuffle result to
   // the vector components that they come from.
-  for (uint32_t component_index = 0;
-       component_index < static_cast<uint32_t>(message_.component_size());
-       component_index++) {
-    uint32_t component = message_.component(component_index);
-    if (component == 0xFFFFFFFF) {
-      // This component is undefined, so move on - but first note that the
-      // overall shuffle result cannot be synonymous with any vector.
-      continue;
-    }
+  if (!transformation_context->GetFactManager()->IdIsIrrelevant(
+          message_.vector1()) &&
+      !transformation_context->GetFactManager()->IdIsIrrelevant(
+          message_.vector2())) {
```

Comment:
As per the following comment, these checks could be moved inside the loop so that we can still deduce facts when one vector is relevant and the other irrelevant.

Reply:
Fixed in 5bea7f6.